harbor/datadog_checks/harbor/harbor.py

```diff
@@ -0,0 +1,121 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from requests import HTTPError
+
+from datadog_checks.base import AgentCheck
+
+from .api import HarborAPI
+from .common import HEALTHY, VERSION_1_5, VERSION_1_8
+
+
+class HarborCheck(AgentCheck):
+    def __init__(self, *args, **kwargs):
+        super(HarborCheck, self).__init__(*args, **kwargs)
+
+        # Prevent the use of Basic Auth using `username` and `password` from the config file.
+        del self.http.options['auth']
+
+        # Keep a single session in order to submit the session id cookie for each request.
+        self.http.persist_connections = True
+
+    def _check_health(self, api, base_tags):
+        """Submits service checks for Harbor individual components."""
+        if api.harbor_version >= VERSION_1_8:
+            health = api.health()
+            for el in health['components']:
+                component_status = AgentCheck.OK if el['status'] == HEALTHY else AgentCheck.CRITICAL
+                tags = base_tags + ['component:{}'.format(el['name'])]
+                self.service_check('harbor.status', component_status, tags=tags)
```

Comment:
It'd be nice to have constants for those.

Reply:
👍